app/models/spree/order.rb

```diff
@@ -104,6 +104,8 @@ def states
     before_save :update_shipping_fees!, if: :complete?
     before_save :update_payment_fees!, if: :complete?
 
+    after_save_commit :save_default_addresses
```

Comment:
You know what? You could move that method to the service as well:
```suggestion
    after_save_commit DefaultAddressUpdater
```

And then you add:
```
class DefaultAddressUpdater
  def self.after_save_commit(order)
      return unless order.save_bill_address || order.save_ship_address

      new(order).call
  end
end
```

Well, that's just moved. It could be written better.

Anyway, just an idea to remove code from the Order class.

Reply:
Oooo nice. I like it.

Reply:
Added :+1: